src/praisonai/tests/unit/cli/test_config_single_source.py

```
@@ -0,0 +1,269 @@
+
+    def test_permissions_from_config_rule_list(self):
+        from praisonai.cli.commands.run import _permissions_from_config
+
+        class Cfg:
+            permissions = {
+                "default": "ask",
+                "rules": [
+                    {"pattern": "bash:git *", "action": "allow"},
+                    {"pattern": "bash:rm *", "action": "deny"},
+                ],
+            }
```

Ruff (0.15.18)

[warning] Mutable default value for class attribute

(RUF012)